sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleDeviceIdCore.java

```diff
@@ -225,6 +248,17 @@ public String getID() {
             }
         }
 
+        /**
+         * Sets current device id to the new one.
```

Comment:
how about this:
Sets device ID according to the device ID Type.
If previous ID was Developer Supplied sets it without merge, otherwise with merge.